docs/reference/transform/checkpoints.asciidoc

```diff
@@ -5,50 +5,70 @@
-field in the source indices. The {transform} uses the values in that field to 
-synchronize the source and destination indices.
- 
-. Updates the destination index (the {dataframe}) with the changed entities.
+The {transform} searches to see which entities or time buckets have changed
+between the last and the new checkpoint. The {transform} uses the values to
+synchronize the source and destination indices with fewer operations than a
+full re-run.
+
+. Updates the destination index (the {dataframe}) with the changes
 +
 --
-The {transform} applies changes related to either new or changed entities to the 
-destination index. The set of changed entities is paginated. For each page, the 
-{transform} performs a composite aggregation using a `terms` query. After all 
-the pages of changes have been applied, the checkpoint is complete.
+The {transform} applies changes related to either new or changed entities or
+time buckets to the destination index. The set of changes can be paginated. The
+{transform} performs a composite aggregation like for the run once case, however
+injects query filters based on step 2, to reduce the amount work. After all
+changes have been applied, the checkpoint is complete.
 --
 
 This checkpoint process involves both search and indexing activity on the
 cluster. We have attempted to favor control over performance while developing
-{transforms}. We decided it was preferable for the {transform} to take longer to 
-complete, rather than to finish quickly and take precedence in resource 
-consumption. That being said, the cluster still requires enough resources to 
-support both the composite aggregation search and the indexing of its results. 
+{transforms}. We decided it was preferable for the {transform} to take longer to
+complete, rather than to finish quickly and take precedence in resource
+consumption. That being said, the cluster still requires enough resources to
+support both the composite aggregation search and the indexing of its results.
 
 TIP: If the cluster experiences unsuitable performance degradation due to the
 {transform}, stop the {transform} and refer to <<transform-performance>>.
 
+[discrete]
+[[ml-transform-checkpoint-heuristics]]
+== Change Detection Heuristics
+
+When transform runs in continuous mode it updates the documents in the
+destination index as new data comes in. Transform uses a set of heuristics
+called change detection to update the destination index with fewer operations.
+
+As an example, assume you are grouping on hostnames. Change detection will detect
+which hostnames have changed, e.g. host `A`, `C` and `G` and only update documents
+with those hosts but not documents that store information about host `B`, `D`, ...
+
+Another Heuristic can be applied for time buckets if you use a `date_histogram` to
+group by time buckets. Change detection will detect which time buckets have changed
+and only update those.
+
 [discrete]
```

Comment:
Another improvement idea:

It would be good to explain a checkpointing best practice: using ingest timestamps (requires 7.11) instead of timestamps that are coming from outside. However I am not sure if this belongs here or in the put transform documentation (or both)

Reply:
this is already mentioned in create transform